### PR TITLE
Route zfs package into recovery environment build

### PR DESCRIPTION
**Pull request by pcd1193182**
This PR moves the recovery-environment package into the kernel build chain so that it can take the zfs build products to install them into the recovery environment.  This PR is intended to merged in conjunction with https://github.com/delphix/recovery-environment/pull/3.  It has been tested by running ab-pre-push , and by loading the built recovery environment onto a test VM and verifying that the system can import pools while in the recovery environment.

### CODE REVIEW

File: packages/recovery-environment/config.sh [hunk 1]
```diff
@@ -19,8 +19,15 @@
 DEFAULT_PACKAGE_GIT_URL="https://github.com/delphix/recovery-environment.git"
 DEFAULT_PACKAGE_VERSION=1.0.0
 
+ZFS_DEB_PATH="$TOP/packages/zfs/tmp/artifacts"
 function prepare() {
+	if [ ! "$(ls -A $ZFS_DEB_PATH)" ]; then
```
Comment (jwk404) [inline packages/recovery-environment/config.sh:24]: Curious why you chose `ls -A` over `[ ! -d ...]` here?

Comment (pcd1193182) [reply packages/recovery-environment/config.sh:24]: -d checks if the path exists and is a directory; it doesn't check if the directory is empty or not, which is what this command does.

File: packages/recovery-environment/config.sh [hunk 2]
```diff
@@ -19,8 +19,15 @@
...
+		logmust "$TOP/buildpkg.sh" zfs
+	fi
+
 	logmust install_build_deps_from_control_file
+	logmust mkdir "$WORKDIR/repo/external-debs/"
+	logmust rsync -a "$ZFS_DEB_PATH/" "$WORKDIR/repo/external-debs/"
```
Comment (pzakha) [inline packages/recovery-environment/config.sh:30]: I think we can limit this to `.deb` files only as there's a bunch of other unrelated files in the artifacts

Comment (pcd1193182) [reply packages/recovery-environment/config.sh:30]: I tried that, and for some reason it didn't work. Originally this was `cp -r "$ZFS_DEB_PATH/*.deb" "$WORKDIR/repo/external-debs/"` but it consistently failed with "No such file or directory" and I wasn't able to determine why. `ls`ing the path revealed plenty of files matching the blob.

Comment (pzakha) [reply packages/recovery-environment/config.sh:30]: Ah, that's because you quoted the * so it takes it literally. Try this:
```
cp -r "$ZFS_DEB_PATH/"*.deb "$WORKDIR/repo/external-debs/"
```
